Composer/packages/client/src/recoilModel/dispatchers/project.ts

```diff
@@ -281,8 +292,12 @@ export const projectDispatcher = () => {
     try {
       await flushExistingTasks(callbackHelpers);
       set(botOpeningState, true);
-      await openRootBotAndSkillsByProjectId(callbackHelpers, projectId);
-
+      const { requiresMigrate } = await openRootBotAndSkillsByProjectId(callbackHelpers, projectId);
```

Comment:
The reason I had this handled externally is that navigating inside these methods causes issues with the calling function.

That is, where we call openRootBot*, we are doing lots of stuff AFTER it resolves.  So if we navigate half-way, the remainder of that stuff is left in a weird place.

Hence, returning this status allows us to STOP the process and navigate rather than continue...